packages/firestore/src/util/bundle_reader.ts

```diff
@@ -175,25 +174,29 @@ export class BundleReader {
    * If reached end of the stream, returns a null.
    */
   private async readLength(): Promise<Uint8Array | null> {
-    let position = this.indexOfOpenBracket();
-    while (position < 0) {
+    let position: number;
+    while ((position = this.indexOfOpenBracket()) < 0) {
       const done = await this.pullMoreDataToBuffer();
       if (done) {
-        if (this.buffer.length === 0) {
-          return null;
-        }
-        position = this.indexOfOpenBracket();
-        // Underlying stream is closed, and we still cannot find a '{'.
-        if (position < 0) {
-          this.raiseError(
-            'Reached the end of bundle when a length string is expected.'
-          );
-        }
-      } else {
-        position = this.indexOfOpenBracket();
+        break;
       }
     }
 
+    // Broke out of the loop because underlying stream is closed, and there
+    // happens to be no more data to process.
+    if (this.buffer.length === 0) {
+      return null;
+    }
+
+    position = this.indexOfOpenBracket();
```

Comment:
You can drop `position` from the loop above and assign it here as a const.

Reply:
Done.